modules/caddyhttp/fileserver/staticfiles.go

```diff
@@ -476,7 +483,12 @@ func (fsrv *FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request, next c
 	// that errors generated by ServeContent are written immediately
 	// to the response, so we cannot handle them (but errors there
 	// are rare)
-	http.ServeContent(w, r, info.Name(), info.ModTime(), file.(io.ReadSeeker))
+
+	if fsrv.PreferPrecompressed {
+		http.ServeContent(w, r, compressedInfo.Name(), compressedInfo.ModTime(), file.(io.ReadSeeker))
+	} else {
+		http.ServeContent(w, r, info.Name(), info.ModTime(), file.(io.ReadSeeker))
+	}
```

Comment:
I think we can just do `info = compressedInfo` to simplify this to a single call to `http.ServeContent`

Reply:
Agreed. Pushed a new commit. Refactored to use only one variable info info instead of 2 separate variables, info and compressedInfo.